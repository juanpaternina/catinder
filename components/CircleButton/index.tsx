import { StyleSheet, Pressable, TouchableOpacity } from "react-native";

type Props = {
  onPress?: () => void;
  children: React.ReactNode;
};

export default function CircleButton(props: Props) {
  return (
    <TouchableOpacity onPress={props.onPress} style={styles.container}>
      {props.children}
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    width: 54,
    height: 54,
    borderRadius: 27,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#FFF",
    boxShadow: "0px 10px 16px 0px #BFBFC04D",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
});
