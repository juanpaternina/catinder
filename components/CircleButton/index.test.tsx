import { Text } from "react-native";
import { render, screen } from "@testing-library/react-native";
import CircleButton from "./index";

describe("CircleButton", () => {
  it("renders correctly", () => {
    render(
      <CircleButton onPress={() => {}}>
        <Text>Test</Text>
      </CircleButton>
    );
    expect(screen.getByText("Test")).toBeTruthy();
    expect(screen).toMatchSnapshot();
  });
});
