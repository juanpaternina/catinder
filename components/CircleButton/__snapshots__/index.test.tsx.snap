// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 27,
      "boxShadow": "0px 10px 16px 0px #BFBFC04D",
      "height": 54,
      "justifyContent": "center",
      "opacity": 1,
      "width": 54,
    }
  }
>
  <Text>
    Test
  </Text>
</View>
`;
